### PR TITLE
Adding blocking API to simulate blocking calls.

### PR DESCRIPTION
**Pull request by mrugankvalat**

### CODE REVIEW

**Comment by mrugankvalat** [comment]
I need to squash all the commits - will do after approval and before merging 😄

File: grpc-messenger-service/src/main/java/com/kucuk/server/service/BlockService.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    public BlockService() {
+    }
+
+    @Override
+    public void blockingCall(BlockingCallRequest request,
+                      StreamObserver<BlockingCallResponse> responseObserver) {
+        if (request.getBlockPeriodInMilliSeconds() > 0) {
+            try {
+                Thread.sleep(request.getBlockPeriodInMilliSeconds());
```
Comment (ilkucuk) [inline grpc-messenger-service/src/main/java/com/kucuk/server/service/BlockService.java]: we are not supposed to be blocking on the api right? https://stackoverflow.com/questions/57329994/how-can-i-manage-blocking-code-inside-grpc-java-server-method

File: grpc-messenger-service/src/main/java/com/kucuk/server/MessageServer.java [hunk 2]
```diff
@@ -59,12 +73,33 @@ private void blockUntilShutdown() throws InterruptedException {
         }
     }
 
+    private ManagedChannel getChannel() throws SSLException {
```
Comment (ilkucuk) [inline grpc-messenger-service/src/main/java/com/kucuk/server/MessageServer.java]: getChannel is too generic, and this (createChannelForBlockingCallToTheOtherService) might be too long. :)

File: grpc-messenger-service/src/main/java/com/kucuk/server/service/MessageService.java [hunk 3]
```diff
@@ -1,81 +1,106 @@
...
+import java.util.concurrent.ExecutionException;
 
 public class MessageService extends MessageServiceGrpc.MessageServiceImplBase {
+    private final ManagedChannel channel;
 
-    @Override
-    public void createMessage(CreateMessageRequest request,
-                             StreamObserver<CreateMessageResponse> responseObserver) {
+    public MessageService(ManagedChannel channel) {
+        this.channel = channel;
```
Comment (ilkucuk) [inline grpc-messenger-service/src/main/java/com/kucuk/server/service/MessageService.java]: channel is too generic of a name?

**Pull request merged**